src/main/java/org/wso2/carbon/inbound/salesforce/poll/SalesforceStreamData.java

```diff
@@ -17,211 +17,214 @@
-                    }
-                    boolean success = message.isSuccessful();
-                    if (!success) {
-                        String error = (String) message.get("error");
-                        if (StringUtils.isNotEmpty(error)) {
-                            handleException("Error during HANDSHAKE: " + error);
-                        }
-                        Exception exception = (Exception) message.get("exception");
-                        if (exception != null) {
-                            handleException("Exception during HANDSHAKE: ", exception);
-                        }
-                    }
-                }
-            });
-
-            client.getChannel(Channel.META_CONNECT).addListener(new ClientSessionChannel.MessageListener() {
-                public void onMessage(ClientSessionChannel channel, Message message) {
-                    log.info("META_CONNECT with the Channel for Salesforce Streaming Endpoint" + message);
-                    boolean success = message.isSuccessful();
-                    boolean clientConnection = client.isConnected();
-                    if (!success && !clientConnection) {
-                        channel.unsubscribe();
-                        client.disconnect();
-                        log.info("Waiting to Connect with Salesforce Streaming API......");
-                        try {
-                            makeConnect();
-                        } catch (IOException e) {
-                            handleException("Error during make the client: " + e.getMessage(), e);
-                        }
-                        String error = (String) message.get("error");
-                        if (StringUtils.isNotEmpty(error)) {
-                            handleException("Error during CONNECT: " + error);
-                        }
-                    }
-                }
-            });
-
-            client.getChannel(Channel.META_SUBSCRIBE).addListener(new ClientSessionChannel.MessageListener() {
-                public void onMessage(ClientSessionChannel channel, Message message) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("META_SUBSCRIBE with the Channel for Salesforce Streaming Endpoint" + message);
-                    }
-                    boolean success = message.isSuccessful();
-                    if (!success) {
-                        String error = (String) message.get("error");
-                        if (StringUtils.isNotEmpty(error)) {
-                            handleException("Error during SUBSCRIBE: " + error);
-                        }
-                    }
+        String str;
+        BufferedReader bufferedReader = null;
+        try {
+            bufferedReader = new BufferedReader(new FileReader(filePath));
+            str = bufferedReader.readLine();
+            bufferedReader.close();
```

Comment:
close the connection within a finally block

Reply:
fixed